rosbag2_cpp/src/rosbag2_cpp/info.cpp

```diff
@@ -67,9 +69,88 @@ struct service_req_resp_info
   std::unordered_map<client_id, sequence_set, client_id_hash> request;
   std::unordered_map<client_id, sequence_set, client_id_hash> response;
 };
+
+struct action_service_req_resp_info
+{
+  service_req_resp_info send_goal_service;
+  service_req_resp_info cancel_goal_service;
+  service_req_resp_info get_result_service;
+};
+
+inline void calculate_message_counts(
+  std::unordered_map<client_id, sequence_set, client_id_hash> & message_map,
+  size_t & count)
+{
+  count = 0;
+  for (auto & [client_id, message_list] : message_map) {
+    count += message_list.size();
+  }
+}
+
+using action_analysis =
+  std::unordered_map<std::string, std::shared_ptr<action_service_req_resp_info>>;
+
+inline void summary_action_service_info(
```

Comment:
Please consider renaming this function for clarity
```suggestion
inline void update_action_service_info_with_num_req_resp(
```